dri3_ws.c

```diff
@@ -189,7 +196,7 @@ static struct driws_buffer *create_buffer(struct driws_drawable *drawable)
 
 #ifdef DRI3WS_USE_DUMB
 	struct drm_mode_create_dumb creq = { };
-	creq.width = drawable->width;
+	creq.width = buffer_width;
```

Comment:
Did you test both dumb and gbm buffers? I haven't tried this, but it looks a bit odd.

My first thought was, why not align the drawable->width, already in create_buffers(). But perhaps the drawable should reflect the requested size (but I don't really know).

Looking at create_buffers, the gbm and dumb paths look different. With this patch, both cases use buffer_width for the allocation, but then gbm case uses gbm_bo_get_width(bo) as width, and dumb case uses drawable->width as width. Overall there are multiple uses of different widths in the function, and it wasn't right away clear to me whether rounded or non-rounded value should be used.

Reply:
Thanks for your comments, I will work on them over the next couple of days. It would be good to have a gbm setup to test with, but I couldn't get it to work - the linked TI libgbm repo seems to be missing the gbm_bo_map that dri3wsegl needs.

Reply:
Well... That's odd. I'm pretty sure I have used gbm with dri3wsegl. But indeed ti-libgbm doesn't have gbm_bo_map. So, I'm officially baffled. It's been a few years since I worked on dri3wsegl, and I have to say I don't recall what this could be about...

Reply:
The 1.17 blobs do provide a more up to date gbm that I think has gbm_bo_map but I don't think that uses WSEGL at all (although I am vaguely looking into some kind of similar-ish shim between DRI3 and gbm EGL to get those blobs working with X11)

Reply:
I have decided to change the gbm path to use drawable->width for the width downstream too, I think the width should reflect the original size as stride is the correct place for the rounded width. Unfortunately I still haven't been able to test the gbm path.